src/base/settingsstorage.h

```diff
@@ -62,4 +62,34 @@ public slots:
     mutable QReadWriteLock m_lock;
 };
 
+template<typename T>
+class StorableValue
+{
+public:
+    explicit StorableValue(const std::string &keyName, const QVariant &defaultValue = QVariant())
```

Comment:
Let me express it in another way:
`SettingsStorage::storeValue(const QString &key, const QVariant &value);`
it uses `QVariant` so I just forward it to here. So another dev won't have to track back to `SettingsStorage` to find it only accept `QVariant` type.